src/layers/h3-hexagon-layer/h3-hexagon-layer.js

```diff
@@ -224,7 +224,7 @@ export default class HexagonIdLayer extends Layer {
 
     allData.forEach((d, index) => {
       const id = getHexId(d);
-      if (typeof id !== 'string' || !id.length) {
+      if (typeof id !== 'string' || !id.length || !h3IsValid(id)) {
```

Comment:
if `h3IsValid(id)` will cover `id !== 'string' || !id.length` case, then we don't need to check the other 2

Reply:
Internally, it calls https://github.com/uber/h3-js/blob/master/lib/h3core.js#L84 which looks like it should account for those - if the string is of the wrong length it should return something like `[0, 0]` which will be invalid.